File: Lottery.cpp
/* Lottery.cpp determines whether a lotto player wins a prize.

Written by M. R. Diaz on December 20th, 2020
*******************************************************************************************************************************************/

#include <iostream>    // cin, cout, <<, >>
#include <string>      // string 
#include <ctime>       // for time function
#include <cstdlib>     // for rand and strand functions

using namespace std;

int main()
{
	string progPurpose = "\nThis program randomly generates lottery numbers which are checked to find out winners.\n";
	cout << progPurpose << endl;

	srand(time(0));
	int lottery = (rand() % 100);    // generates lottery numbers

	// prompt the player to enter a guess for the numbers
	cout << "\nEnter your lottery pick (two digits): ";
	int guess;
	cin >> guess;

	int lotteryDigit1 = (lottery / 10);  // get digits from lottery
	int lotteryDigit2 = (lottery % 10);

	int guessDigit1 = (guess / 10);    // get digits from lotto player
	int guessDigit2 = (guess % 10);

	cout << "\nThe lottery number is " << lottery << endl;

	if (guess == lottery)
	{
		cout << "\nExact match: You win $10K! " << endl;
	}
	else if (guessDigit2 == lotteryDigit1 && guessDigit1 == lotteryDigit2)
	{
		cout << "\nMatch all digits: You win $3K " << endl;
	}
	else if (guessDigit1 == lotteryDigit1
		|| guessDigit1 == lotteryDigit2
		|| guessDigit2 == lotteryDigit1
		|| guessDigit2 == lotteryDigit2)
	{

		cout << "\nMatch one digit: You win $1K " << endl;
	}
	else
	{
		cout << "\nSorry, no match.\n" << endl;

		system("pause");   // console output stays on 
		return(0);

			// End of program
	}


}

File: Lottery.java
/*Lottery.java determines whether a lottery player wins a prize.
 * 
 * Input: User chooses digits to play
 * Output: Program determines whether a prize is won.
 * 
 * December 19th, 2020  
 *************************************************************************************************/

package simplePrograms;

import java.util.*;   //Gets the Scanner class from the package (library) java.util

public class Lottery 
{
	public static void main(String[] args) 
	{
		String progPurpose = "\nThis program ramdomly generates lottery numbers which are checked to find out winners.\n";
		System.out.println(progPurpose);
		
		int lottery = (int)(Math.random() * 100);   //generates random numbers 
		
		Scanner keyboard = new Scanner(System.in);  // user guesses numbers to play
		System.out.print("\nEnter your lottery pick (two digits): ");
		int guess = keyboard.nextInt();
		
		int lotteryDigit1 = (lottery / 10);  // get digits from lottery
		int lotteryDigit2 = (lottery % 10);
		
		int guessDigit1 = (guess / 10);      // digits from user
		int guessDigit2 = (guess % 10);
		
		System.out.println("\nThe lottery number is " +lottery);
		
		if (lottery == guess)      // criteria to win any prize
		System.out.print("Exact match: You win $10K!");
		else if (guessDigit2 == lotteryDigit1 && guessDigit2 == lotteryDigit2)
		System.out.println("Match all digits: You win $3K!");
		else if (guessDigit1 == lotteryDigit1 || guessDigit1 == lotteryDigit2 
				|| guessDigit2 == lotteryDigit1 || guessDigit2 == lotteryDigit2)
		System.out.println("You match one digit: You win $1K!");
		else
		System.out.println("\nSorry, no match.");
		
		keyboard.close(); //prevents resource leakage
		
		// End of program
	}

}
